### PR TITLE
Update documentation and defaults for memory overcommit

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Reestablish documentation for the memory overcommit. Set `memory_usage_overcommit_max_wait_microseconds` to `200` by default.

cc @alexey-milovidov @alesapin

### CODE REVIEW

File: docs/en/operations/settings/memory-overcommit.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Memory overcommit
+
+Memory overcommit is an experimental technique intended to allow to set more flexible memory limits for queries.
+
+The idea of this technique is to introduce settings which can represent guaranteed amount of memory a query can use.
+When memory overcommit is enabled and the memory limit is reached ClickHouse will select the most overcommitted query and try to free memory by killing this query.
+
+When memory limit is reached any query will wait some time during atempt to allocate new memory.
```
Comment (nickitat) [inline docs/en/operations/settings/memory-overcommit.md:8]: attempt

File: docs/en/operations/settings/memory-overcommit.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+If timeout is passed and memory is freed, the query continues execution.
+Otherwise an exception will be thrown and the query is killed.
+
+Selection of query to stop or kill is performed by either global or user overcommit trackers depending on what memory limit is reached.
+If overcommit tracker can't choose query to stop, MEMORY_LIMIT_EXCEEDED exception is thrown.
+
+## User overcommit tracker
+
+User overcommit tracker finds a query with the biggest overcommit ratio in the user's query list.
+Overcommit ratio for a query is computed as number of allocated bytes divided by value of `memory_overcommit_ratio_denominator` setting.
```
Comment (nickitat) [inline docs/en/operations/settings/memory-overcommit.md:18]: why it is not (allocated - guaranteed)?

**Review by nickitat — APPROVED** [review]

File: docs/en/operations/settings/memory-overcommit.md (hunk 1 above)
Comment (novikd) [reply docs/en/operations/settings/memory-overcommit.md:8]:
```suggestion
When memory limit is reached any query will wait some time during attempt to allocate new memory.
```

File: docs/en/operations/settings/memory-overcommit.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+If `memory_overcommit_ratio_denominator` for the query is equals to zero, overcommit tracker won't choose this query.
+
+Waiting timeout is set by `memory_usage_overcommit_max_wait_microseconds` setting.
+
+**Example**
+
+```sql
+SELECT number FROM numbers(1000) GROUP BY number SETTINGS max_guaranteed_memory_usage=4000, memory_usage_overcommit_max_wait_microseconds=500
```
Comment (novikd) [inline docs/en/operations/settings/memory-overcommit.md:27]:
```suggestion
SELECT number FROM numbers(1000) GROUP BY number SETTINGS memory_overcommit_ratio_denominator=4000, memory_usage_overcommit_max_wait_microseconds=500
```

**Comment by novikd** [comment]
@Mergifyio update

**Pull request merged**